internal/policy/verify.go

```diff
@@ -686,7 +604,30 @@ func verifyEntry(ctx context.Context, repo *git.Repository, policy *State, attes
 	return nil
 }
 
-func verifyTagEntry(ctx context.Context, repo *git.Repository, policy *State, entry *rsl.ReferenceEntry) error {
+func verifyAttestation(ctx context.Context, verifiers []*Verifier, commitObj *object.Commit, authorizationAttestation *sslibdsse.Envelope) (bool, string, error) {
+	// No verifiers => no restrictions for the git namespace
+	if len(verifiers) == 0 {
+		return true, "", nil
+	}
+	for _, verifier := range verifiers {
+		err := verifier.Verify(ctx, commitObj, authorizationAttestation)
+		if err == nil {
+			// Signature verification succeeded
+			return true, verifier.Name(), nil
+		} else if !errors.Is(err, ErrVerifierConditionsUnmet) {
+			// Unexpected error
+			return false, "", err
+		}
+		// Haven't found a valid verifier, continue with next
+	}
+	return false, "", nil
+}
+
+func verifyTagEntry(ctx context.Context, repo *git.Repository, policy *State, entry *rsl.ReferenceEntry, target string) error {
+	if _, _, err := rsl.GetLatestReferenceEntryForRefBefore(repo, target, entry.GetID()); err == nil {
```

Comment:
Depending on when this is invoked, this could still let things fall through. What if we're verifying the first tag entry and the duplicate is after this one? Also, this change may belong in a separate PR to keep things scoped. For this PR, let's keep the focus on fixing up VerifyRelativeForRef and verifyEntry.